mistralrs-core/src/pipeline/paths.rs

```diff
@@ -381,12 +382,17 @@ pub(crate) fn get_chat_template(
     } else {
         panic!("Expected chat template file to end with .json, or you can specify a tokenizer model ID to load the chat template there. If you are running a GGUF model, it probably does not contain a chat template.");
     };
-
     let mut template: ChatTemplate = match chat_template_ovrd {
         Some(chat_template) => {
             // In this case the override chat template is being used. The user must add the bos/eos/unk toks themselves.
             info!("Using literal chat template.");
             let mut template = ChatTemplate::default();
+            if chat_template.contains("<|end|>") {
```

Comment:
I think we should avoid such hardcoded cases generally, but was there a bug being fixed here?

Reply:
> I think we should avoid such hardcoded cases generally, but was there a bug being fixed here?

Yes, it fixes a bug related to running Phi3 GGUF models, where the end-of-sequence token ("<|end|>") is not properly parsed. If there is no external `chat_template.json` or `config.json`, we need to parse both the BOS and EOS tokens from the GGUF file (the literal chat template). I think there's a more elegant way to handle this situation, and I'm willing to replace the hardcoded one.

Reply:
I think it's better to raise an issue #1012

Reply:
Sounds good. I'll take a look at that issue, we can make a fix outside of this PR!